src/ClrFacade/Internal.cs

```diff
@@ -554,6 +554,8 @@ public static object ConvertRSharpGenericValue(RSharpGenericValue argument)
             return argument.GetArray<float>();
          case RSharpValueType.StringArray:
             return argument.GetArray<string>();
+         case RSharpValueType.ObjectArray:
+            return convertToObjectArguments(argument.Value, argument.Size);

```

Comment:
If we get an array type, then we already have this method that converts arrays of intPtr to object arrays.